### PR TITLE
feat(version): display verions of @angular/* and @ngtools/*

### LINKED ISSUE #3589: `ng version` should show more information
List of versions that would be useful:

- Angular version itself (all packages starting with `@angular/`
- All our own published packages (starting with `@angular-cli/` and `@ngtools/`)
- TypeScript

Other ideas are welcome

### PR DESCRIPTION
**Pull request by Brocco**
Fixes #3589

### CODE REVIEW

**Comment by Brocco** [comment]
@hansl PR updated to resolve the lint and build issues.

File: packages/angular-cli/commands/version.ts [hunk 1]
```diff
@@ -14,12 +14,19 @@ const VersionCommand = Command.extend({
   }],
 
   run: function (options: any) {
-    const versions: any = process.versions;
+    let versions: any = process.versions;
```
Comment (hansl) [inline packages/angular-cli/commands/version.ts]: Why the change?

File: packages/angular-cli/commands/version.ts [hunk 2]
```diff
@@ -33,15 +40,54 @@ const VersionCommand = Command.extend({
...
 
     for (const module of Object.keys(versions)) {
-      if (options.verbose || alwaysPrint.indexOf(module) > -1) {
+      let isRoot = false;
+      roots.forEach(root => {
+        if (module.startsWith(root)) {
+          isRoot = true;
+        }
+      });
+      if (options.verbose || alwaysPrint.indexOf(module) > -1 || isRoot) {
```
Comment (hansl) [inline packages/angular-cli/commands/version.ts]: `isRoot` could be set to `roots.some(root => module.startsWith(root))` and it would be simpler.

File: packages/angular-cli/commands/version.ts [hunk 3]
```diff
@@ -33,15 +40,54 @@ const VersionCommand = Command.extend({
...
+        if (key && key.startsWith(prefix)) {
+          modules[key] = this.getVersion(key);
+        }
+      }
+    }
+    return modules;
+  },
+
+  getVersion: function(moduleName: string): string {
+    const modulePkg = require(path.resolve(
```
Comment (hansl) [inline packages/angular-cli/commands/version.ts]: Could you at least try locally to `nom link` angular itself and run `ng version`? Just wanna make sure that works.

File: packages/angular-cli/commands/version.ts [hunk 4]
```diff
@@ -33,15 +40,54 @@ const VersionCommand = Command.extend({
...
+    const modules: any = {};
+    if (pkg.dependencies) {
+      for (let key of Object.keys(pkg.dependencies)) {
+        if (key && key.startsWith(prefix)) {
+          modules[key] = this.getVersion(key);
+        }
+      }
+    }
+    if (pkg.devDependencies) {
+      for (let key of Object.keys(pkg.devDependencies)) {
```
Comment (hansl) [inline packages/angular-cli/commands/version.ts]: This whole function could be simplified to:

```typescript
Object.keys(pkg.dependencies || {})
  .concat(Object.keys(pkg.devDependencies || {}))
  .filter(depName => depName && depName.startsWith(prefix))
  .forEach(depName => modules[key] = this.getVersion(key));
```

What do you think?

File: packages/angular-cli/commands/version.ts [hunk 5]
```diff
@@ -33,15 +40,54 @@ const VersionCommand = Command.extend({
...
+          isRoot = true;
+        }
+      });
+      if (options.verbose || alwaysPrint.indexOf(module) > -1 || isRoot) {
         this.printVersion(module, versions[module]);
       }
     }
   },
 
+  getModules: function(pkg: any, prefix: string): any {
```
Comment (hansl) [inline packages/angular-cli/commands/version.ts]: `getDependencyVersions` or `getVersionsOfDependencies` might be a better name, what do you think?

File: packages/angular-cli/commands/version.ts (hunk 1 above)
Comment (Brocco) [inline packages/angular-cli/commands/version.ts]: Because later on I have this...
```
versions = Object.assign(versions, this.getModules(projPkg, root));
```
which disallows the use of const

File: packages/angular-cli/commands/version.ts (hunk 5 above)
Comment (Brocco) [inline packages/angular-cli/commands/version.ts]: I like `getDependencyVersions` and went with that.

File: packages/angular-cli/commands/version.ts (hunk 2 above)
Comment (Brocco) [inline packages/angular-cli/commands/version.ts]: yes, cleaner

File: packages/angular-cli/commands/version.ts (hunk 4 above)
Comment (Brocco) [inline packages/angular-cli/commands/version.ts]: I like this as well, updated this function, but had to fix a minor issue with what you have here

File: packages/angular-cli/commands/version.ts (hunk 3 above)
Comment (Brocco) [inline packages/angular-cli/commands/version.ts]: I tested locally via npm link for both inside and outside of CLI projects.

**Comment by hansl** [comment]
LGTM! Thanks!
